src/main/java/apoc/graph/inverse/builder/RelationshipBuilder.java

```diff
@@ -0,0 +1,43 @@
+package apoc.graph.inverse.builder;
+
+import apoc.graph.util.FromConfig;
+import org.neo4j.graphdb.Direction;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Relationship;
+import org.neo4j.graphdb.RelationshipType;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class RelationshipBuilder {
+
+    private FromConfig config;
+
+    public RelationshipBuilder(FromConfig config) {
+        this.config = config;
+    }
+
+    public Collection<Relationship> buildRelation(Node parent, Node child, String relationName) {
+//        String relationName = child.getProperty(config.getLabelFieldName()).toString().toUpperCase();
+
+        RelationshipType type = RelationshipType.withName(relationName.toUpperCase());
+
+        //check if already exists
+        Iterable<Relationship> relationships = parent.getRelationships(Direction.OUTGOING, type);
+
+        //find only relation between parent and child node
+        List<Relationship> rels = StreamSupport.stream(relationships.spliterator(), false)
+                .filter(rel -> rel.getOtherNode(parent).getId() == child.getId())
```

Comment:
getOtherNode(parent).equals(child)

Reply:
Would it not make more sense to start from the child? in terms of number of relationships to look at.

Reply:
with incoming then of course.

Reply:
Yes, I agree with you but it's implemented in this way because in case of VirtualNode the inverse relationship is not registered into the child node so if we start the discovery from the child we don't have any reference to the parent. I'll disambiguate the two use case and:

 * the VirtualNode I'll leave as is
 * the real node I'll start the discovery from the child

Or do you prefer registering the inverse relationship into the virtual node?